### ISSUE TITLE
1.20.1: Crash on world creation

### ISSUE THREAD

**Issue by SCsupercraft** [thedead2/customadvancements#14]
**Versions**
- Minecraft Version: 1.20.1
- Forge Version: 47.2.20
- Custom Advancements Version: 1.20.1-7.0.8

**Describe the bug**
Game crashes on world creation

**Steps to reproduce**
1. Create world
2. That's it

**Screenshots and Logs**

[crash-2024-05-10_19.08.09-client.txt](https://github.com/thedead2/customadvancements/files/15278501/crash-2024-05-10_19.08.09-client.txt)
[latest.log](https://github.com/thedead2/customadvancements/files/15278513/latest.log)
![Screenshot 2024-05-10 191404](https://github.com/thedead2/customadvancements/assets/103709272/3faf978b-f768-49d3-9e39-e1ec71bc1b94)

**Additional context**

Here is the pack it occurred on 
[Create Factory Builder-Alpha 0.5.0.zip](https://github.com/thedead2/customadvancements/files/15278574/Create.Factory.Builder-Alpha.0.5.0.zip)

**Comment by SCsupercraft** [comment]
@thedead2 it also seems to be the reason you get this screen when loading a world (happens every other time)

![image](https://github.com/thedead2/customadvancements/assets/103709272/00d63b87-2215-4991-bdfb-922fb20bf826)


Here is the full log
[latest.log](https://github.com/thedead2/customadvancements/files/15285405/latest.log)

**Comment by thedead2** [comment]
Hi,
thanks for reaching out. I can't tell directly from the log, what is going wrong. My best guess is, that the mod is trying to remove an advancement (by its id) which doesn't exist. Maybe it's a misspelled resource location in the blacklist/ whitelist section of the config. Please check those entries and make sure they are all resource locations of existing advancements. You can try to remove all entries one by one to check wich entry is causing the crash.
My other guess is that it has to do with the advancement datapack you've installed. I actually never tested the mod with datapacks so I can't really say if it's compatible or not. So if the above method of removing advancement's one by one from the blacklist/ whitelist doesn't work, try to remove the advancement datapack.
Either way, this shouldn't cause the game to crash, so I will try to change the code so your game won't crash if something like this happens and rather will give you an detailed description what went wrong. Hope this helps. 

PS: Please don't post code as a comment, it's almost impossible to read that way. Attaching the log is better.

**Comment by SCsupercraft** [comment]
Yep, you were right I must of removed some of the modpacks advancements for that build since they were not finished but clearly forgot the remove them from the whitelist. Thanks for your help, everything works properly now!

**Comment by thedead2** [comment]
Hi that sounds great. Glad I could help. I will keep this issue open until I've released a fix so the game won't crash directly and instead tries to tell you what went wrong instead.
